Standards/WCAG2AAA/Sniffs/Principle1/Guideline1_3/1_3_5.js

```diff
@@ -0,0 +1,388 @@
+ * +--------------------------------------------------------------------+
+ * | IMPORTANT: Your use of this Software is subject to the terms of    |
+ * | the Licence provided in the file licence.txt. If you cannot find   |
+ * | this file please contact Squiz (www.squiz.com.au) so we may        |
+ * | provide you a copy.                                                |
+ * +--------------------------------------------------------------------+
+ *
+ */
+
+_global.HTMLCS_WCAG2AAA_Sniffs_Principle1_Guideline1_3_1_3_5 = {
+    /**
+   * Determines the elements to register for processing.
+   *
+   * Each element of the returned array can either be an element name, or "_top"
+   * which is the top element of the tested code.
+   *
+   * @returns {Array} The list of elements.
+   */
+    register: function() {
+        return ["_top", "input", "select", "textarea"];
+    },
+
+    /**
+   * Checks that the values of the autocomplete attribute are values listed in the HTML 5.2 spec.
+   *
+   * @param {DOMNode} element The element registered.
+   */
+    checkValidAttributes: function(element) {
+        var valid_attributes = [
+            "additional-name",
+            "address-level1",
+            "address-level2",
+            "address-level3",
+            "address-level4",
+            "address-line1",
+            "address-line2",
+            "address-line3",
+            "bday",
+            "bday-year",
+            "bday-day",
+            "bday-month",
+            "billing",
+            "cc-additional-name",
+            "cc-csc",
+            "cc-exp",
+            "cc-exp-month",
+            "cc-exp-year",
+            "cc-family-name",
+            "cc-given-name",
+            "cc-name",
+            "cc-number",
+            "cc-type",
+            "country",
+            "country-name",
+            "current-password",
+            "email",
+            "family-name",
+            "fax",
+            "given-name",
+            "home",
+            "honorific-prefix",
+            "honorific-suffix",
+            "impp",
+            "language",
+            "mobile",
+            "name",
+            "new-password",
+            "nickname",
+            "off",
+            "on",
+            "organization",
+            "organization-title",
+            "pager",
+            "photo",
+            "postal-code",
+            "sex",
+            "shipping",
+            "street-address",
+            "tel-area-code",
+            "tel",
+            "tel-country-code",
+            "tel-extension",
+            "tel-local",
+            "tel-local-prefix",
+            "tel-local-suffix",
+            "tel-national",
+            "transaction-amount",
+            "transaction-currency",
+            "url",
+            "username",
+            "work"
+        ];
+
+        var valuesStr = element.getAttribute("autocomplete");
+        var values = valuesStr.split(" ");
```

Comment:
This line needs some defensive code to handle undefined attribute value
```suggestion
        if (typeof valuesStr !== 'string') { 
            return;
        }
        var values = valuesStr.split(" ");
```

Reply:
Actually it looks like it's not strictly necessary from the code that calls this, but I think it's better to have it than not.